### PR TITLE
feat(ecs-patterns): pass securityGroups to ScheduledFargateTask

### LINKED ISSUE #5213: (ecs-patterns): use existing Security Group with ScheduledFargateTask
## :question: General Issue
Unable to find a way to specify an existing security group when creating task schedules for fargate tasks using aws cdk

### The Question

When defining an ECS Task Schedule, I can't seem to find a way of specifying an existing security group. Any pointers on where this can be configured using aws cdk?

In the code snippet below, you'll see I am able to create a cron, specify the docker image to schedule and create the schedule itself by specifying the existing cluster and vpc. However, there is no option to specify an existing security group... Is it possible to specify an existing security group?

``` python
schedule_cron = scaling.Schedule.cron(minute=manifest['schedule']['minute'], 
                                                hour=manifest['schedule']['hour'], 
                                                day=manifest['schedule']['day'], 
                                                month=manifest['schedule']['month'], 
                                                year=manifest['schedule']['year'])

image_option = ecs_patterns.ScheduledFargateTaskImageOptions(image=img, 
                                                cpu=manifest["resources"]["cpu"], 
                                                memory_limit_mib=manifest["resources"]["memory"],
                                                log_driver=ecs.AwsLogDriver(log_group=log_group,
                                                                            stream_prefix=manifest["app_name"]),
                                                secrets=secrets,
                                                environment= env)

schedule_pattern = ecs_patterns.ScheduledFargateTask(self, f"scheduledtask{app_name}", 
                                                schedule= schedule_cron, scheduled_fargate_task_image_options=image_option, cluster=cluster, 
                                                desired_task_count=manifest["replica_count"], vpc=vpc)

```

When i do a `cdk synth` i can see that cdk is trying to generate a new security group with some outbound and inbound rules. Instead, i would like to pass an existing security group that's been defined with relevant organization access policies. Is it supported? 

### Environment

  - **CDK CLI Version:** 1.15.0
  - **Module Version:** 1.15.0
  - **OS:** Windows
  - **Language:** Python

### Other information

https://stackoverflow.com/questions/59067514/aws-cdk-ecs-task-scheduling-specify-existing-securitygroup

### PR DESCRIPTION
**Pull request by crazymykl**
closes #5213

This is a minimal approach.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by RajarsiGit — APPROVED** [review]

**Review by RajarsiGit — APPROVED** [review]

**Review by RajarsiGit — APPROVED** [review]

**Review by RajarsiGit — APPROVED** [review]

**Comment by fongie** [comment]
just struggled with this in my project: this would be a nice feature

**Comment by RajarsiGit** [comment]
> just struggled with this in my project: this would be a nice feature

Hi, @fongie can you share you code please, wherein you have overcome this issue using some way? Thanks

**Review by RajarsiGit — APPROVED** [review]
Looks good to me

**Comment by fongie** [comment]
> > just struggled with this in my project: this would be a nice feature
> 
> Hi, @fongie can you share you code please, wherein you have overcome this issue using some way? Thanks

Hi! Sure, but I warn you it's not pretty and there might be a better way too :)
```typescript
  private addSecurityGroupToScheduledTask(group: SecurityGroup, task: ScheduledFargateTask) {
    const rule = task.eventRule as any as CfnRule
    const targets = rule.targets as any as RuleTargetConfig[]
    const config = targets[0].ecsParameters!.networkConfiguration as any
    const groups = config.awsVpcConfiguration.securityGroups as Array<string>
    groups.push(group.securityGroupId)
  }
```

This adds the security group you pass in to the method to the security groups of the task. Then, you can configure connections via the security group (like allowfrom, to, etc). Note that it also keeps the default security group.

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/scheduled-fargate-task.ts [hunk 1]
```diff
@@ -33,6 +34,13 @@ export interface ScheduledFargateTaskProps extends ScheduledTaskBaseProps {
    * @default Latest
    */
   readonly platformVersion?: FargatePlatformVersion;
+
+  /**
+   * Existing security groups to use for your service.
+   *   *
+   * @default 'A new security group is created'
```
Comment (SoManyHs) [inline packages/@aws-cdk/aws-ecs-patterns/lib/fargate/scheduled-fargate-task.ts:41]:
```suggestion
   * @default A new security group is created.
```

**Review by SoManyHs — COMMENTED** [review]
This is great! Do you think you could also add this to the ScheduledEc2Task as well?

**Comment by SoManyHs** [comment]
I went ahead and added the change to support Ec2 patterns on top of your work here: https://github.com/aws/aws-cdk/pull/15096

**Comment by crazymykl** [comment]
Nice. I was having some trouble figuring out how to unit test that case.

**Comment by SoManyHs** [comment]
Just curious, is it critical that you can reference the securityGroup as an output property on the ScheduledFargateTask after it is created? I.e. https://github.com/aws/aws-cdk/pull/14220/files#diff-7ebbca721fecf8f6c66398af515f1d264cbdf3116d34ef912aab19a55e06994aR39-R43?

**Comment by SoManyHs** [comment]
Resolved by #15096

**Pull request closed without merging**